core/templates/domain/exploration/ParamSpecsObjectFactory.ts

```diff
@@ -101,10 +101,14 @@ export class ParamSpecs {
    *    consumption.
    */
   toBackendDict(): ParamSpecsBackendDict {
-    var paramSpecsBackendDict = {};
-    this.forEach((paramName, paramSpec) => {
-      paramSpecsBackendDict[paramName] = paramSpec.toBackendDict();
-    });
+    var paramSpecsBackendDict: ParamSpecsBackendDict = {};
+    this.forEach(
+      (
+          paramName: string | number,
+          paramSpec: { toBackendDict: () => ParamSpecBackendDict; }
```

Comment:
Is this correct? Can you please explain.

Reply:
No, this should be of type `ParamSpec ` corrected it!